### PR TITLE
Validate that keyed sets have struct encoding with key

### PR DESCRIPTION
**Pull request by charleso**

### CODE REVIEW

File: ivory-operation/src/test/scala/com/ambiata/ivory/operation/ingestion/DictionaryImportValidateSpec.scala [hunk 1]
```diff
@@ -21,6 +21,8 @@ class DictionaryImportValidateSpec extends Specification with ScalaCheck { def i
    is invalid with a virtual definition with an invalid filter     $virtualInvalidFilter
    is invalid with a virtual definition with an invalid expression $virtualInvalidExpression
    is invalid when there are more then Short.MaxValue features     $numFeaturesInvalid
+   is invalid when the encoding for keyed set is not a struct      $keyedSetInvalidEncoding
+   is invalid when the keyed set key is missing                    $keyedSetMissingKey
```
Comment (markhibberd) [inline ivory-operation/src/test/scala/com/ambiata/ivory/operation/ingestion/DictionaryImportValidateSpec.scala]: Should also be mandatory I would think, or does this cover that as well?

Comment (charleso) [inline ivory-operation/src/test/scala/com/ambiata/ivory/operation/ingestion/DictionaryImportValidateSpec.scala]: > Should also be mandatory I would think, or does this cover that as well?

Good point, and no this doesn't.

File: ivory-core/src/test/scala/com/ambiata/ivory/core/arbitraries/Arbitraries.scala [hunk 2]
```diff
@@ -51,7 +51,7 @@ trait Arbitraries {
     Arbitrary(GenIdentifier.dictionary)
 
   implicit def EncodingArbitrary: Arbitrary[Encoding] =
-    Arbitrary(GenDictionary.encoding)
+    Arbitrary(GenDictionary.mode.flatMap(GenDictionary.encoding))
```
Comment (markhibberd) [inline ivory-core/src/test/scala/com/ambiata/ivory/core/arbitraries/Arbitraries.scala]: :cry: I think this should be like it was, and there should be a `.encoding` and a `.encodingWithMode` or similar on GenDictionary.

Comment (charleso) [inline ivory-core/src/test/scala/com/ambiata/ivory/core/arbitraries/Arbitraries.scala]: > I think this should be like it was, and there should be a .encoding and a .encodingWithMode or similar on GenDictionary.

Yeah, I did wonder about that. Happy to change.

**Comment by markhibberd** [comment]
:+1: but if you feel saddened by the arbitrary cleanliness, I wouldn't be adverse to it being tightened back up.

**Pull request merged**